### PR TITLE
Create comments

### PR DESCRIPTION
**Pull request by Tuyendk4**

### CODE REVIEW

File: app/views/comments/_comment.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<li id="comment-<%= comment.id %>">
+  <%= link_to gravatar_for(comment.user, size: 50), comment.user %>
+  <span class="user"><%= link_to comment.user.name, comment.user %></span>
+  <span class="timestamp">
+    commented <%= time_ago_in_words(comment.created_at) %> ago.
+      <% if current_user?(comment.user) %>
+      <%= link_to t(".edit"), "#editForm#{comment.id}", data: {toggle: "collapse"}, aria: {expanded: "false", controls: "#editForm#{comment.id}"} %>
+      <%= link_to t(".delete"), comment, method: :delete, remote: true,
+        data: { confirm: "You sure?" } %>
```
Comment (manhcuongdtbk) [inline app/views/comments/_comment.html.erb]: i18n

File: app/assets/javascripts/comments.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (manhcuongdtbk) [inline app/assets/javascripts/comments.coffee]: remove redundant file

File: config/routes.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@
   resources :categories
   resources :ratings, only: :update
   resources :relationships, only: %i(create destroy)
+  resources :comments
```
Comment (manhcuongdtbk) [inline config/routes.rb]: only ... ?

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        end
+      end
+    end
+  
+    def edit
+      @comment = current_user.comments.find_by id: params[:id]
+    end
+  
+    def update
+      if @comment.update_attributes(comment_params)
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: remove()

File: app/views/comments/_comment.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    commented <%= time_ago_in_words comment.created_at %> ago.
+      <% if current_user? comment.user %>
+      <%= link_to t(".edit"), "#editForm#{comment.id}", data: {toggle: "collapse"}, aria: {expanded: "false", controls: "#editForm#{comment.id}"} %>
+      <%= link_to t(".delete"), comment, method: :delete, remote: true,
+        data: { confirm: t(".confirm") } %>
+    <% end %>
+  </span>
+  <br />
+  <span class="timestamp">
+    updated <%= time_ago_in_words comment.updated_at %> ago.
```
Comment (chintk) [inline app/views/comments/_comment.html.erb]: i18n

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        respond_to do |format|
+          format.html { redirect_to book_path(@comment.book_id),
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: redirect_to @comment.book

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    
+    def new
+      @comment = Comment.new
+    end
+  
+    def create
+      @comment = Comment.new comment_params
+      if @comment.save
+        respond_to do |format|
+          format.html { redirect_to book_path(@comment.book_id),
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: use do end

File: app/controllers/comments_controller.rb [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    end
+  
+    def edit
+      @comment = current_user.comments.find_by id: params[:id]
+    end
+  
+    def update
+      if @comment.update_attributes comment_params
+        respond_to do |format|
+          format.html { redirect_to book_path(@comment.book_id),
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: do end

File: app/models/book.rb [hunk 9]
```diff
@@ -7,7 +7,7 @@ class Book < ApplicationRecord
   has_many :categories, through: :book_categories
   has_many :book_authors
   has_many :authors, through: :book_authors
-  default_scope ->{order(created_at: :desc)}
+  scope :order_sort, -> { order(created_at: :desc) }
```
Comment (chintk) [inline app/models/book.rb]: remove()

File: app/models/comment.rb [hunk 10]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :book
+  scope :order_sort, -> { order(created_at: :desc) }
```
Comment (chintk) [inline app/models/comment.rb]: remove()

File: app/controllers/comments_controller.rb [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+  
+    def edit
+    end
+  
+    def update
+      if @comment.update_attributes comment_params
+        respond_to do |format|
+          format.html do 
+            redirect_to book_path(@comment.book_id), 
+            notice: t(".flash_notice")
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: indent

File: app/controllers/comments_controller.rb [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+      @comment = Comment.new
+    end
+  
+    def create
+      @comment = Comment.new comment_params
+      if @comment.save
+        respond_to do |format|
+          format.html do 
+            redirect_to book_path(@comment.book_id), 
+            notice: t(".flash_notice")
```
Comment (chintk) [inline app/controllers/comments_controller.rb]: indent

**Pull request merged**